src/SFML/Audio/SoundSource.cpp

```diff
@@ -25,155 +25,349 @@
 ////////////////////////////////////////////////////////////
 // Headers
 ////////////////////////////////////////////////////////////
-#include <SFML/Audio/ALCheck.hpp>
 #include <SFML/Audio/SoundSource.hpp>
 
-#if defined(__APPLE__)
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-#endif
+#include <miniaudio.h>
+
+#include <algorithm>
+
 
 namespace sf
 {
 // NOLINTBEGIN(readability-make-member-function-const)
+// NOLINTBEGIN(clang-analyzer-cplusplus.PureVirtualCall)
 ////////////////////////////////////////////////////////////
-SoundSource::SoundSource()
-{
-    alCheck(alGenSources(1, &m_source));
-    alCheck(alSourcei(m_source, AL_BUFFER, 0));
-}
-
-
-////////////////////////////////////////////////////////////
-SoundSource::SoundSource(const SoundSource& copy)
+SoundSource::SoundSource(const SoundSource& copy) : AudioResource(copy)
 {
-    alCheck(alGenSources(1, &m_source));
-    alCheck(alSourcei(m_source, AL_BUFFER, 0));
-
     setPitch(copy.getPitch());
+    setPan(copy.getPan());
     setVolume(copy.getVolume());
+    setSpatializationEnabled(copy.isSpatializationEnabled());
     setPosition(copy.getPosition());
+    setDirection(copy.getDirection());
+    setCone(copy.getCone());
+    setVelocity(copy.getVelocity());
+    setDopplerFactor(copy.getDopplerFactor());
     setRelativeToListener(copy.isRelativeToListener());
     setMinDistance(copy.getMinDistance());
+    setMaxDistance(copy.getMaxDistance());
+    setMinGain(copy.getMinGain());
+    setMaxGain(copy.getMaxGain());
     setAttenuation(copy.getAttenuation());
 }
 
 
 ////////////////////////////////////////////////////////////
-SoundSource::~SoundSource()
+void SoundSource::setPitch(float pitch)
 {
-    alCheck(alSourcei(m_source, AL_BUFFER, 0));
-    alCheck(alDeleteSources(1, &m_source));
+    if (auto* sound = static_cast<ma_sound*>(getSound()); sound)
```

Comment:
```suggestion
    if (auto* sound = static_cast<ma_sound*>(getSound()))
```

Ditto for all other occurrences

Reply:
Applied this suggestion myself.